packages/style-engine/class-wp-style-engine-renderer.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * WP_Style_Engine_Renderer
+ *
+ * Compiles and renders styles from a store of CSS rules.
+ *
+ * @package Gutenberg
+ */
+
+if ( class_exists( 'WP_Style_Engine_Renderer' ) ) {
+	return;
+}
+
+/**
+ * Compiles and renders styles from a store of CSS rules.
+ *
+ * @access private
+ */
+class WP_Style_Engine_Renderer {
```

Comment:
In terms of naming, I was wondering what you and @ramonjd think about potentially calling this an `Optimizer` instead of a renderer. Something that's been on my mind lately is how the naming of the style engine and its component parts can set expectations of what it will (or might be able to do) and `Renderer` suggests to me that it might also handle other things than optimizing CSS and outputting CSS rules.

I don't feel too strongly about it, though, if you think Renderer is a better fit!

Reply:
That sounds fair. This processing is the last step before we publish/render the 'optimized' styles to the page.

There's probably a semantic argument for any word here 😄 , but I agree I think we could choose a different term just to be clear... especially if, in the context of the style engine, we use 'render' to mean something like page/browser rendering. 👍 

`Optimizer` sounds fine, or `Processor/[insert synonym]`, given that this class appears to be a good place to add further pre-render tasks.

Reply:
It depends on how we envision the Renderer object...
This PR merely adds the 1st step to what I hope the Rendered will become.
Ideally, the Renderer with combine the CSS from all rules in a store (done in this PR), optimize the CSS (somewhat done in this PR), and then enqueue the generated styles (pending)
Enqueueing the CSS is not handled in this PR because we'll need to thing that part of the object some more and figure out the details for all cases. Since this is about inline styles, they will most likely be added using `wp_add_inline_styles`, which means they'll need to be "attached" to a pre-existing stylesheet. But not all stores will use the same stylesheet... So we need to think how to implement that part some more before committing something.
With the above in mind, I called the object `Renderer`... the CSS optimizing might be a relatively big part of the code here, but is just a small part of the final envisioned implementation.

We could however call this one `Processor`, and then create a separate `Renderer` if and when the need for it arises 🤔 

Thoughts?

Reply:
Actually, the more I think of this, the more `Processor` makes sense... 
We'll be processing and generating and final CSS. How that CSS is used/enqueued can be left to the individual implementations.
I'll go ahead and rename the object 👍